packages/cdktf/lib/terraform-variable.ts

```diff
@@ -51,6 +51,11 @@ export abstract class VariableType {
   }
 }
 
+export interface TerraformVariableValidationConfig {
+  readonly condition: any;
```

Comment:
I'm unsure whether we could further narrow down this type?
Maybe allow `string | IResolvable`? This will effectively have the same behaviour as `any` in languages like e.g. Java (treating it as `Object`) but will carry more type information in TypeScript and Python.
If we expect a `can` function to be used at the root of the expression, we could even go with only `IResolvable` and drop `string`.

Reply:
`Expression` is pretty close to the correct type (technically any), though there are limits to what can be contained in the expression.

Reply:
I have added the `Expression` as @jsteinich said.